cl/corpus_importer/management/commands/harvard_opinions.py

```diff
@@ -356,59 +379,251 @@ def parse_harvard_opinions(reporter, volume, make_searchable):
+
+    :param docket_number: Case docket number
+    :return: A stripped down docket number.
+    """
+
+    docket_number = re.sub("Department.*", "", docket_number)
+    docket_number = re.sub("Nos?. ", "", docket_number)
+    return docket_number
+
+
+def clean_body_content(case_body: str) -> str:
+    """Strip all non alphanumeric characters
+
+    :param case_body: Opinion text
+    :return:Opinion text with only alphanumeric characters
+    """
+    soup = BeautifulSoup(case_body, "lxml")
+    return re.sub(r"[^[a-zA-Z]", "", soup.text)
+
+
+def match_based_text(
+    case_body: str, possible_cases: List, docket_number: str
+) -> Optional[OpinionCluster]:
+    """Compare CL text to Harvard content to establish duplicates
+
+    :param case_body: Harvard case body to compare
+    :param possible_cases: List of opinions to check against
+    :param docket_number: The docket number
+    :return: OpinionCluster or None
+    """
+    harvard_characters = clean_body_content(case_body)
+    for case in possible_cases:
+        cl_case_body = get_opinion_content(case)
+        cl_characters = clean_body_content(cl_case_body)
+        if (
+            len(harvard_characters) / len(cl_characters) < 0.3
+            or len(harvard_characters) / len(cl_characters) > 3
+        ):
+            # Content too dissimilar in length to compare
+            continue
+
+        percent_match = compare_documents(harvard_characters, cl_characters)
+        if percent_match < 45:
+            continue
+        if len(harvard_characters) < 500:
+            if not percent_match in range(98, 102):
+                continue
+            clean_docket = clean_docket_number(docket_number)
+            if clean_docket not in case.docket.docket_number:
+                continue
+        return case
+    return None
+
+
+def find_previously_imported_cases(
+    court_id: str,
+    date_filed: date,
+    case_body: str,
+    docket_number: str,
+) -> Optional[OpinionCluster]:
+    """Check if opinion is in Courtlistener
+
+    :param court_id: Court ID
+    :param date-filed: The date filed
+    :param case_body: Date of opinion
+    :param docket_number: The docket number
+    :return:
+    """
+    possible_cases = OpinionCluster.objects.filter(
+        date_filed=date_filed,
+        docket__court_id=court_id,
+    ).order_by("id")
+    month = timedelta(days=31)
+    broad_search = OpinionCluster.objects.filter(
+        date_filed__range=[date_filed - month, date_filed + month],
+        docket__court_id=court_id,
+    ).order_by("id")
+
+    match = match_based_text(case_body, possible_cases, docket_number)
+    if not match:
+        possible_cases = [
+            case for case in broad_search if case.date_filed != date_filed
+        ]
+        match = match_based_text(case_body, possible_cases, docket_number)
+    return match
+
+
+def compare_documents(harvard_characters: str, cl_characters: str) -> int:
+    """Compare Harvard text to CL opinion text
```

Comment:
I think I see what you're doing here, but it's finicky and I bet some tests would flush out a bug or two and be easy to whip up. Would you mind?

Reply:
Sounds like a plan.

Reply:
I added a test to do comparison of similar small opinions fixed a bug inside the system that would let multiple overlapping text cause slight inaccuracies in the final percent match.